world/petitions/petitions_commands.py

```diff
@@ -8,9 +8,179 @@
 from server.utils.prettytable import PrettyTable
 from world.petitions.forms import PetitionForm
 from world.petitions.exceptions import PetitionError
-from world.petitions.models import BrokeredSale, Petition
+from world.petitions.models import BrokeredSale, Petition,WantedAd
+from web.character.models import (Clue, SearchTag)
 
 
+
+
+class CmdMatchmaker(ArxCommand):
+    """
+    Creates or searches wanted ads
+    
+        Usage:
+    -Viewing:
+        matchmaker
+        matchmaker/all
+    -Editing
+        matchmaker/active #Toggles on/off
+        matchmaker/blurb <text>
+        matchmaker/playtimes <start>,<end> #Assume EST
+        matchmaker/clues <clue1>,<clue2>
+        matchmaker/tags <tag1>,<tag2>
+        matchmaker/seeking <marriage>,<plot>,<protege>,<sponsor> #any
+    """
+    key ="matchmaker"
+    help_category = "social"
+    aliases =["mm"]
+    
+    def display_matches(self):
+        my_ad = self.caller.dompc.wanted_ad
+        ads = WantedAd.objects.filter(active=True)
+        matches_list = []
+        matched_list = []
+        for ad in ads:
+            matches = ad.match(self.caller.dompc)
+            if matches and (ad.TIME in matches):
+                matches_list.append(matches)
+                matched_list.append(ad)
+        table = PrettyTable(["Name", "Gender", "Age", "Matches","Playtime", "Blurb"])
```

Comment:
You may wanna use EvTable rather than PrettyTable due to line wrap issues. Otherwise you run into that pretty fast unless you do a lot of truncation of the columns. Line-wrap looks pretty hideous in PrettyTable and makes it hard to read, so it's not just a cosmetic issue, unofrtunately.

I prefer using PrettyTable where I know there won't be wrap issues due to speed - EvTable starts to have marked performance issues when there's a good number of matches, like beyond a few dozen. That's why for a few commands I have EvTable when there's a relatively small number of rows, but otherwise use PrettyTable with heavy truncation for a large number of matches.

Reply:
I had been wondering why you were using two different tables.